### DISCUSSION TITLE
How to make CotEditor the default text editor?

### DISCUSSION THREAD

**Discussion by zcysxy** [coteditor/CotEditor#1360] [Q&A]
I have tried the top two methods in this thread [replace-text-edit-as-the-default-text-editor](https://apple.stackexchange.com/questions/123833/replace-text-edit-as-the-default-text-editor), but they didn't help.

**Comment by 1024jp** [comment]
Hmm, this is something not about CotEditor but the macOS system in general.

Generally, the user change the default application for a specific file type by setting it in the inspector in Finder (just like in [this reply](https://apple.stackexchange.com/a/175525)). Select CotEditor in the "Open with" pop up menu and then press the "Change All" button:

<img width="650" alt="Screen Shot 2022-03-19 at 11 32 38" src="https://user-images.githubusercontent.com/1165044/159103490-55d00634-6c15-4ce6-b6fa-00190bc92ece.png">

In addition, in case the file mapping can be broken for some reasons, you'll need to reset the database of Raunch Services.

**Comment by zcysxy** [comment]
@1024jp Thanks for the reply. Yeah, this changes the default app for a single extension. What I intend to do is to open any text file in CotEditor. I think my problem is that there is another app, BBEdit, competing with CotEditor. Currently in terminal

```bash
open -t anyTextFile
```

opens the file in CotEditor, which means CotEditor is indeed the default editor. However, when I double-click the file, it still opens in BBEdit.

**Comment by AlexRusbridge** [comment]
Unfortunately, there's no real way of natively changing the default application for text files on mass.

I know some applications like [IINA](https://github.com/iina/iina) have a button within their settings that programmatically changes the default application for all video common file types. Maybe this could be a feature to implement in the future?

Meanwhile, you can use an application called [SwiftDefaultApps](https://github.com/Lord-Kamina/SwiftDefaultApps) that should give you a similar function within a settings preference pane:

<img width="780" alt="Screenshot 2022-03-27 at 9 26 26 am" src="https://user-images.githubusercontent.com/6761721/160273356-1043ae6a-ad33-430b-8f40-f8c03e87218f.png">

**Reply by eyalrin** [reply]
What about this method?
[https://notes.zander.wtf/change-default-text-editor-mac](url)

We just need to know the text to enter in the end that matches for CotEditor...

**Reply by heckman** [reply]
I think you're referring to the bundle identifier.
For CotEditor it's `com.coteditor.CotEditor`.

**Comment by ei-zascha** [comment]
You can use [duti](https://github.com/moretension/duti) to do this:
`brew install duti`

To set CotEditor as the default for all plain text files:
`duti -s com.coteditor.CotEditor public.plain-text all`

To confirm:
`duti -x text`

**Reply by heyitsazar** [reply]
Thank you so much!
